dbms/src/main/java/org/polypheny/db/processing/AbstractQueryProcessor.java

```diff
@@ -135,90 +139,146 @@
 import org.polypheny.db.transaction.TableAccessMap.Mode;
 import org.polypheny.db.transaction.TableAccessMap.TableIdentifier;
 import org.polypheny.db.transaction.TransactionImpl;
-import org.polypheny.db.type.ArrayType;
-import org.polypheny.db.type.ExtraPolyTypes;
 import org.polypheny.db.type.PolyType;
 import org.polypheny.db.util.ImmutableIntList;
 import org.polypheny.db.util.Pair;
-import org.polypheny.db.util.Util;
 import org.polypheny.db.view.MaterializedViewManager;
 import org.polypheny.db.view.MaterializedViewManager.TableUpdateVisitor;
 import org.polypheny.db.view.ViewManager.ViewVisitor;
 
 
 @Slf4j
-public abstract class AbstractQueryProcessor implements QueryProcessor {
+public abstract class AbstractQueryProcessor implements QueryProcessor, ExecutionTimeObserver {
 
     protected static final boolean ENABLE_BINDABLE = false;
     protected static final boolean ENABLE_COLLATION_TRAIT = true;
     protected static final boolean ENABLE_ENUMERABLE = true;
     protected static final boolean CONSTANT_REDUCTION = false;
     protected static final boolean ENABLE_STREAM = true;
-
     private final Statement statement;
 
+    private final Map<Integer, Long> scanPerTable = new HashMap<>(); // scanId  -> tableId //Needed for Lookup
```

Comment:
We should improve the comment for this map and explain that it is required for allowing plans with multiple physical placements of the same logical table. We should also make sure that this does not introduce unintended side effects, since rewriting a plan creates copies of the relNodes which have different ids.